### PR TITLE
Play song when it's doubleclicked (TrackRow)

### PR DESCRIPTION
**Pull request by olafur-palsson**
By injecting a prop into TrackPopupContainer via TrackRow.

Sorry for verbose commit, I accidentally did auto-indent with vim. :(

### CODE REVIEW

**Comment by nukeop** [comment]
Can you run eslint on this? Might take care of the formatting.

**Comment by olafur-palsson** [comment]
Shit I'm horrible at this git business. I'm going to study the docs more. Thanks for putting up with the mess. :)

File: app/components/TrackRow/index.js [hunk 1]
```diff
@@ -43,6 +49,11 @@ class TrackRow extends React.Component {
     );
   }
 
+  playTrack () {
+    console.log(this)
```
Comment (nukeop) [inline app/components/TrackRow/index.js]: Could you remove this?

File: app/containers/TrackPopupContainer/index.js [hunk 2]
```diff
@@ -23,13 +23,33 @@ const TrackPopupContainer = props => {
...
+          name: props.title,
+          thumbnail: props.thumb
+        });
+      actions.selectSong(0);
+      actions.startPlayback();
+    }
+  };
+
+  if (doubleClick) {
+    playSong();
```
Comment (nukeop) [inline app/containers/TrackPopupContainer/index.js]: Anything that dispatches actions should stay out of the render method, and only do so in response to some event.

**Comment by nukeop** [comment]
Sorry it took me a few days to get back to you. Your solution works, but it should be structured differently. The TrackRow should dispatch the actions necessary to start playing the song, since it responds to the double click event. Like I commented in the code, action logic should stay out of render methods, due to the risk of being launched at unexpected times when React decides to re-render the component.

I'm thinking that maybe we should get rid of the popup appearing when you click anywhere on the track row, maybe there should be a dedicated button (with a vertical ellipsis) at its edge, so we could get clean double clicks without opening the popup.

**Comment by olafur-palsson** [comment]
Sorry, my master branch was polluted. I will migrate this to another pull request.

**Pull request closed without merging**